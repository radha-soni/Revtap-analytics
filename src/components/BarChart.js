import React from "react";
import { Bar } from "react-chartjs-2";

import Paper from "@material-ui/core/Paper";
import "./chart.css";

export default function BarChart({ customers }) {
  const created = [];
  const orderCount = [];

  if (customers) {
    customers.forEach((element) => {
      created.push(element.created.slice(0, 10));
    });
    removeDuplicate(created);
  }
  function removeDuplicate(created) {
    created.splice(0, created.length, ...new Set(created));
    created.forEach((element) => {
      let value = 0;
      customers.forEach((date) => {
        if (element === date.created.slice(0, 10)) {
          value = value + Number(date.orders);
        }
      });
      orderCount.push(value);
    });
  }
  const state = {
    labels: created,

    datasets: [
      {
        label: "Orders",
        backgroundColor: "#1976d2",

        borderWidth: 2,
        data: orderCount,
        barThickness: 10,
      },
    ],
  };
  return (
    <Paper elevation={3} className="bar">
      <div>
        <p>Orders Count</p>
      </div>

      <Bar
        data={state}
        width={600}
        height={400}
        options={{
          title: {
            display: false,
            text: "Orders Count",
            fontSize: 20,
          },
          scales: {
            offset: false,

            xAxes: [
              {
                type: "time",
                distribution: "linear",
                gridLines: {
                  display: false,
                },
              },
            ],
          },
          legend: {
            display: false,
          },
          maintainAspectRatio: false,
          responsive: false,
        }}
      />
    </Paper>
  );
}
