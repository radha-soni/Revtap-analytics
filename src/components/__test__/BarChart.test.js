import React from "react";
import renderer from "react-test-renderer";
import BarChart from "../BarChart.js";
test("bar chart renders correctly", () => {
  const chart = renderer.create(<BarChart />);
  let bar = chart.toJSON();
  expect(bar).toMatchSnapshot();
});
