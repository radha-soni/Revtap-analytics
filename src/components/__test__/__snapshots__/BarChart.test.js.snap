// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bar chart renders correctly 1`] = `
<div
  className="MuiPaper-root bar MuiPaper-elevation3 MuiPaper-rounded"
>
  <div>
    <p>
      Orders Count
    </p>
  </div>
</div>
`;
